pkg/restore/restore_test.go

```diff
@@ -676,6 +676,45 @@ func TestResetMetadataAndStatus(t *testing.T) {
 	}
 }
 
+func TestObjectsAreEqual(t *testing.T) {
+	tests := []struct {
+		name        string
+		backupObj   *unstructured.Unstructured
+		clusterObj  *unstructured.Unstructured
+		expectedErr bool
+		expectedRes bool
+	}{
+		{
+			name:        "objects are already equal",
+			backupObj:   NewTestUnstructured().WithName("obj").WithArkLabel("test").Unstructured,
+			clusterObj:  NewTestUnstructured().WithName("obj").Unstructured,
+			expectedErr: false,
+			expectedRes: true,
+		},
+		{
+			name:        "objects reset correctly",
+			backupObj:   NewTestUnstructured().WithName("obj").WithArkLabel("test").Unstructured,
+			clusterObj:  NewTestUnstructured().WithMetadata("blah", "foo").WithName("obj").Unstructured,
+			expectedErr: false,
+			expectedRes: true,
+		},
+		{
+			name:        "cluster object has no metadata to reset",
+			backupObj:   NewTestUnstructured().WithName("obj").WithArkLabel("test").Unstructured,
+			clusterObj:  NewTestUnstructured().Unstructured,
+			expectedErr: true,
+			expectedRes: false,
+		},
```

Comment:
It might be worth getting the json output from a real cluster of something like a secret or service or pod and putting that in here as a test case. We have a test helper to convert json to Unstructured.